apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/styles/starter-page-templates-editor.scss

```diff
@@ -169,6 +169,7 @@ body.is-fullscreen-mode {
 		overflow: hidden;
 		background-color: #fff;
 		position: relative;
+		transform: translate3d( 0, 0, 0 ); // Fix for Safari rounded border overflow (1/2).
```

Comment:
Love to see the comments for extra context 👍 

I am interested in _why_ this fixes it. Why does it happen and why is this necessary to fix it.

Reply:
> (...) why this fixes it. Why does it happen (...)

Great questions. It seems to be related to how Safari processes [stacking context](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Positioning/Understanding_z_index/The_stacking_context) (which is different than Chrome's). I'm lacking a deep understanding of how exactly that process is applied, so maybe someone could help out here. I bumped into similar issues with Safari before and, unless the explicit `z-index` did not work (which is the case here), the `translate3d(0, 0, 0)` did the trick.  It applies another level to the stacking context that acts as an encapsulator, separating the respective stacking contexts of the involved elements ([source](https://medium.com/@mattcroak718/understanding-the-stacking-context-for-cross-browser-compatibility-2b21db1cf621)). 

I've just realized that the `translateZ(0)` should work as well and would fit better here. Will update.

> (...) why is this necessary to fix it.

This looks like a _hack_, but Safari is known to have issues like this one. We probably could try rebuilding the whole component to try different techniques that wouldn't require forced `transform` to make it look good in Safari, but I'm not sure if it's worth the effort in this case. Does that make sense?

Reply:
For the larger context, I was also seeing (before the fix) other random artifacts, like this one:

![](https://cld.wthms.co/x242XX+)

Reply:
TIL. All good by me. Thanks for the detailed response.